### PR TITLE
Handle mode/uid/gid in FilesManager.put()

### PR DESCRIPTION
**Pull request by keesbos**
The FilesManager misses the capability to set file mode/uid/gid

### CODE REVIEW

File: pylxd/models/container.py [hunk 1]
```diff
@@ -77,7 +77,20 @@ def __init__(self, client, container):
             self._client = client
             self._container = container
 
-        def put(self, filepath, data):
+        def put(self, filepath, data, mode=None, uid=None, gid=None):
+
+            if isinstance(mode, int):
```
Comment (rockstar) [inline pylxd/models/container.py]: This logic block should fall in under `if mode is not None`.

**Comment by rockstar** [comment]
You've got failing tests, and I think you should add tests for this functionality as well.

**Comment by ajkavanagh** [comment]
@keesbos as @rockstar indicated, this needs tests before it can be merged.  Do you need a hand with that?

**Comment by ajkavanagh** [comment]
I pulled your branch and added tests to it as part of #276 PR.  Thanks for the initial work.  I'm closing this now as I incorporated your commits into PR #276.  Thanks again.

**Pull request closed without merging**